fixes #22 -- include path param for launchSingleTop nav condition

sphrak opened the pull request:
This CS makes it so that `launchSingleTop=true` is per unique route + path param. This is different from before, when `launchSingleTop` would not consider the path parameters which would lead to the fact that it was only possible navigate to one route once, regardless of the path parameter.

sphrak commented on buildSrc/src/main/kotlin/Versions.kt:
```diff
@@ -1,7 +1,7 @@
 
 object Versions {
-    const val precompose = "1.3.14"
+    const val precompose = "1.4.0"
```
i just made this up, lemme know if i should remove it

Tlaster replied:
It's better to leave it as it.

Tlaster reviewed (commented):
Thanks for your PR!
It's better to have another option in `NavOptions` as this breaks the current behavior of `launchSingleTop`.
Sorry for the late response, it's still a busy week for me.

sphrak commented:
> Thanks for your PR! It's better to have another option in `NavOptions` as this breaks the current behavior of `launchSingleTop`. Sorry for the late response, it's still a busy week for me.

fair point, i initially wasn't sure if current behavior was intended or not.. but i have reworked it a bit as requested, i also cleaned it up a bit. lemme know what u think, at ur leisure :-)

sphrak commented on precompose/src/commonMain/kotlin/moe/tlaster/precompose/navigation/BackStackManager.kt:
```diff
@@ -88,9 +88,9 @@ internal class BackStackManager(
+            _backStacks.any { it.hasRoute(matchResult.route.route, path, options.includePath) }
         ) {
-            _backStacks.firstOrNull { it.hasRoute(matchResult.route.route) }?.let {
+            _backStacks.firstOrNull { it.hasRoute(matchResult.route.route, path, options.includePath) }?.let {
```
im not quite sure if this check should also be considering `includePath` in the PopUp case at L118

Tlaster replied:
I thinks it's better to check `includePath' at L118 to maintain consistent behaviour.

sphrak replied:
consider it done

Tlaster reviewed (approved):
Thanks! LGTM!

sphrak commented:
> Thanks! LGTM!

my pleasure, lemme know if there is anything else i should do

Tlaster commented:
Thanks! I'm going to merge it!
